src/Services/MovieService.ts

```diff
@@ -1,97 +1,86 @@
-    genre: 'Action & Adventure',
-    image: 'Avenger.png',
-    date: new Date(2004, 8, 2),
-    url: 'Avenger.com',
-    rating: 3.1,
-    runtime: 131,
-    description: 'Avengers: War of Infinity DESCRIPTION',
-  },
-  {
-    id: 4,
-    title: 'Inception',
-    genre: 'Action & Adventure',
-    image: 'Inception.png',
-    date: new Date(2003, 11, 7),
-    url: 'Inception.com',
-    rating: 7.6,
-    runtime: 119,
-    description: ' DESCRIPTION',
-  },
-  {
-    id: 5,
-    title: 'Reservoir dogs',
-    genre: 'Oscar winning Movie',
-    image: 'ReservoirDogs.png',
-    date: new Date(1994, 3, 25),
-    url: 'ReservoirDogs.com',
-    rating: 7.9,
-    runtime: 99,
-    description: 'Reservoir dogs DESCRIPTION',
-  },
-];
+const baseUrl = 'http://localhost:4000/movies';
 
-export class MovieService {
-  getMovies() : Movie[] {
-    return moviesData;
-  }
-
-  sortMovies(moviesToSort: Movie[], sortOrder: SortOptionType) : Movie[] {
-    let sortedMovies;
-    switch (sortOrder) {
-      case SortOptionType.ReleaseDateAsc:
-        sortedMovies = [...moviesToSort].sort((a, b) => a.date < b.date ? -1 : 1);
+export async function getMoviesApi(genres : string[], sortBy : SortOptionType) : Promise<Movie[]> {
+  let data;
+  try {
+    let sort;
+    switch (sortBy) {
+      case SortOptionType.RatingAsc:
+        sort = 'sortOrder=asc&sortBy=vote_average';
         break;
-      case SortOptionType.ReleaseDateDesc:
-        sortedMovies = [...moviesToSort].sort((a, b) => a.date > b.date ? -1 : 1);
+      case SortOptionType.RatingDesc:
+        sort = 'sortOrder=desc&sortBy=vote_average';
         break;
-      case SortOptionType.TitleAsc:
-        sortedMovies = [...moviesToSort].sort((a, b) => a.title < b.title ? -1 : 1);
+      case SortOptionType.ReleaseDateAsc:
+        sort = 'sortOrder=asc&sortBy=release_date';
         break;
-      case SortOptionType.TitleDesc:
-        sortedMovies = [...moviesToSort].sort((a, b) => a.title > b.title ? -1 : 1);
+      case SortOptionType.ReleaseDateDesc:
+        sort = 'sortOrder=desc&sortBy=release_date';
         break;
     }
 
-    return sortedMovies;
+    let url = baseUrl + '?' + sort;
+    let filter;
+    if (genres.length > 0) {
+      filter = '&filter=' + genres.join(',');
+    }
+
+    if (filter) {
+      url+= filter;
+    }
+
+    const response = await window.fetch(url);
+    data = await response.json();
+    if (response.ok) {
+      return data.data;
+    }
+
+    throw new Error(response.statusText);
+  } catch (err) {
+    return Promise.reject(err.message ? err.message : data);
+  }
+}
+
+
+export async function deleteMovieApi(id : number) {
+  try {
+    const url = baseUrl + '/' + id;
+    const response = await window.fetch(url, {method: 'DELETE'});
+    if (response.ok) {
+      return;
+    }
+
+    throw new Error(response.statusText);
+  } catch (err) {
+    return Promise.reject(err.message);
+  }
+};
+
+export async function updateMovieApi(movie : Movie) {

```

Comment:
```suggestion
export async function updateOrCreateMovieApi(movie : Movie) {
```
Because, you use it for creation and updating (if I understand correctly)
Moreover, I would think about using separate calls